### PR TITLE
fix: do not use `posthandler` to handle failed tx sequence increment

### PR DESCRIPTION
**Pull request by beer-1**
# Description

Closes: #XXXX

to handle this issue: https://github.com/cosmos/cosmos-sdk/issues/22396
we changed to sequence decrement be done in x/evm msg handler.

---

## Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title, you can find examples of the prefixes below:

- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] included the necessary unit and integration tests
- [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] confirmed all CI checks have passed

## Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `IncrementSequenceDecorator` for managing sequence numbers in transactions.
	- Added a constant `ContextKeySequenceIncremented` for enhanced context tracking.

- **Bug Fixes**
	- Streamlined sequence number handling to prevent double incrementing during EVM operations.

- **Removals**
	- Removed the `IncrementSequenceDecorator` and associated methods from the `sigverify` and `posthandler` packages, simplifying transaction processing.

- **Tests**
	- Added a new test suite to validate the functionality of the `IncrementSequenceDecorator`.

### CODE REVIEW

File: x/evm/ante/sequence.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	signers, err := sigTx.GetSigners()
+	if err != nil {
+		return sdk.Context{}, err
+	}
+
+	for _, signer := range signers {
+		acc := isd.ak.GetAccount(ctx, signer)
+		if err := acc.SetSequence(acc.GetSequence() + 1); err != nil {
+			panic(err)
+		}
```
[the remark this answers is not in the document]
Comment (beer-1) [reply x/evm/ante/sequence.go:40]: need to follow same rule with cosmos-idk

File: x/evm/ante/sequence.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	// increment sequence of all signers
+	signers, err := sigTx.GetSigners()
+	if err != nil {
+		return sdk.Context{}, err
+	}
+
+	for _, signer := range signers {
+		acc := isd.ak.GetAccount(ctx, signer)
+		if err := acc.SetSequence(acc.GetSequence() + 1); err != nil {
```
[the remark this answers is not in the document]
Comment (beer-1) [reply x/evm/ante/sequence.go:38]: already incremented, so good to skip this check

File: x/evm/keeper/msg_server.go [hunk 3]
```diff
@@ -256,3 +255,23 @@
...
+// To prevent double incrementing the sequence number during EVM execution, we need to decrement it here for create messages.
+func (k *msgServerImpl) handleSequenceIncremented(ctx context.Context, sender sdk.AccAddress, isCreate bool) (context.Context, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// decrement sequence of the sender
+	if isCreate && sdkCtx.Value(evmante.ContextKeySequenceIncremented) != nil {
+		acc := k.accountKeeper.GetAccount(ctx, sender)
+		if err := acc.SetSequence(acc.GetSequence() - 1); err != nil {
+			return ctx, err
+		}
```
[the remark this answers is not in the document]
Comment (beer-1) [reply x/evm/keeper/msg_server.go]: already incremented, so good to skip the check

**Review by Vritra4 — APPROVED** [review]
LGTM

**Pull request merged**